test/transformers/test_kl_div.py

```diff
@@ -0,0 +1,95 @@
+from test.utils import assert_verbose_allclose, supports_bfloat16
+
+import pytest
+import torch
+from torch.nn import KLDivLoss
+
+from liger_kernel.transformers.kl_div import LigerKLDIVLoss
+
+_SHAPE_PARAMS = (
+    "B, T, V",
+    [
+        (1, 4096, 32000),
+        pytest.param(
+            1,
+            4096,
+            128256,
+            marks=pytest.mark.skipif(
+                torch.cuda.get_device_properties(0).total_memory
+                < 36 * 1000 * 1000 * 1000,
+                reason="This test requires a GPU with at least 36GB of memory",
+            ),
+        ),
+        (3, 423, 32000),
+    ],
+)
+
+_DTYPE_PARAMS = (
+    "dtype, atol, rtol",
+    [
+        pytest.param(
+            torch.bfloat16,
+            1e-8,
+            5e-2,
+            marks=pytest.mark.skipif(
+                not supports_bfloat16(), reason="bfloat16 not supported on this GPU"
+            ),
+        ),
+        (torch.float32, 1e-8, 1e-6),
+        (torch.float16, 1e-3, 1e-2),
+    ],
+)
+
+
+def _test_correctness_once(
+    target_kldiv, B, T, V, dtype, atol, rtol, reduction, log_target, is_last=True
+):
+    torch.manual_seed(0)
+    torch_kldiv = KLDivLoss(reduction=reduction, log_target=log_target)
+
+    input = torch.randn(
+        B * T, V, device="cuda", dtype=dtype, requires_grad=True
+    ).log_softmax(dim=-1)
+
+    x1 = input.detach().clone().requires_grad_(True)
+    x2 = input.detach().clone().requires_grad_(True)
+
+    with torch.no_grad():
+        target = torch.randn(B * T, V, device="cuda").softmax(dim=-1)
+
+    output = torch_kldiv(x1, target)
+    output2 = target_kldiv(x2, target)
+    assert_verbose_allclose(output, output2, atol=atol, rtol=rtol)
+
+    if not is_last:
```

Comment:
Out of curiosity, what is the `is_last` here used for?

Reply:
If not is_last, I multiply the result to do more calculations for grad, since if it's the last operation we can skip one multiply in backwards calculation, since the output grads are 1.0

Reply:
add a comment to explain: is_last = is_last_layer?